### PR TITLE
Fix `DEV-19107`: Refine image-sequence component preload behaviors and rotate-to-index handling

### PR DESCRIPTION
**Pull request by cbutcosk**
This PR updates the image-sequence component to resolve DEV-19107 ("Image Sequence component does not respond to rotate-to-index attribute"):
- Refactor `preloadImages()` to accept an `indices` parameter of indices to preload, and to return the promise of all fetch responses
- Create `animationIndex` property for managing animation draws in `performRotation()`
- Fixes an issue where the rotation Call to Action was not displayed
- Updates comments

### CODE REVIEW

File: packages/11ty/_includes/web-components/image-sequence/index.js [hunk 1]
```diff
@@ -54,9 +63,9 @@ class ImageSequence extends LitElement {
   canvasRef = createRef()
 
   /**
-   * @property bufferReady
+   * @property indexesToBuffer
    * 
-   * Returns an array of indexes to buffer
+   * Returns an array of indexes included in `bufferSize` given `index`
```
Comment (mphstudios) [inline packages/11ty/_includes/web-components/image-sequence/index.js:68]: Use [JSDoc returns tag](https://jsdoc.app/tags-returns)
```suggestion
   * @returns an array of indexes included in `bufferSize` given `index`
```

File: packages/11ty/_includes/web-components/image-sequence/index.js [hunk 2]
```diff
@@ -54,9 +63,9 @@ class ImageSequence extends LitElement {
...
    * 
    **/
```
Comment (mphstudios) [inline packages/11ty/_includes/web-components/image-sequence/index.js:70]: Use [JSDoc returns tag](https://jsdoc.app/tags-returns) and `*/` to close doc comments.
```suggestion
   * @returns an array of indexes included in `bufferSize` given `index`
   */ 
```

File: packages/11ty/_includes/web-components/image-sequence/index.js [hunk 3]
```diff
@@ -246,6 +285,12 @@ class ImageSequence extends LitElement {
     }
   }
 
+  /**
+   * @function hideOverlays
+   * 
+   * TODO: Should be removed in favor of dynamic markup (may already be)
+   * 
+   **/
```
Comment (mphstudios) [inline packages/11ty/_includes/web-components/image-sequence/index.js:293]: Use [JSDoc todo tag](https://jsdoc.app/tags-todo)
```suggestion
  /**
   * @function hideOverlays 
   * @todo removed in favor of dynamic markup (may already be)
   */ 
```

**Review by mphstudios — CHANGES_REQUESTED** [review]
@cbutcosk rethinking the various variable names for the array of indexes that are buffered, and wether they are defined using an implicit getter, or as a _computed_ property in `willUpdate`, before merging this fix would be worth while. For example refactoring `indexesToBuffer` to be defined in `static properties` as a `bufferWindow` property and then _computed_ (perhaps using a private `#computeBufferWindow` method) in `willUpdate`;
```
/**
 * @private
 * Computes the window of image indices that are buffered 
 * as one greater to protect against numerical under run
 */
#computeBufferWindow() {
  const nImages = this.images.length
  const index = nImages + this.index - Math.round(this.bufferSize/2))
  this.bufferWindow = Array(this.bufferSize)
    .fill(0)
    .map((_, i) => ((index + i) % nImages)
}
```
Similarly, should `bufferReady` be defined in the `static properties` and _computed_ in `willUpdate`?

File: packages/11ty/_includes/web-components/image-sequence/index.js [hunk 4]
```diff
@@ -311,55 +356,81 @@ class ImageSequence extends LitElement {
...
    **/
-  #preloadImages() {
+  #preloadImages(indices) {
     if (!this.images.some(i => i === null)) { return }
 
-    // Really just making buffer counting ergonomic / readable here
-    // const indexesToBuf = Array(this.bufferSize).fill(0).map( (_,i) => ( this.images.length + this.index + i - Math.round(this.bufferSize/2) ) % this.images.length )
-    this.images.forEach( (image,i) => {
+    const indexesToLoad = indices ?? this.indexesToBuffer
+    const fetches = this.images.map( (image,i) => {
```
Comment (mphstudios) [inline packages/11ty/_includes/web-components/image-sequence/index.js:405]:
```suggestion
    const image = this.images.map((image,i) => {
```
or
```
    const response = this.images.map((image,i) => {
```

File: packages/11ty/_includes/web-components/image-sequence/index.js [hunk 5]
```diff
@@ -311,55 +356,81 @@ class ImageSequence extends LitElement {
...
-        return
+      if ( !indexesToLoad.includes(i) || image !== null ) {
+        return null
       } 
 
       const url = this.imageUrls[i]
-      this.#fetchImage(url,i)
-    })
+      return this.#fetchImage(url,i)
+    }).filter( ftch => ftch )
```
Comment (mphstudios) [inline packages/11ty/_includes/web-components/image-sequence/index.js:413]:
```suggestion
    }).filter((image) => image)
```

File: packages/11ty/_includes/web-components/image-sequence/index.js [hunk 6]
```diff
@@ -311,55 +356,81 @@ class ImageSequence extends LitElement {
...
+
+
+    return Promise.all(fetches)
```
Comment (mphstudios) [inline packages/11ty/_includes/web-components/image-sequence/index.js:416]:
```suggestion
    return Promise.all(images)
```
or 
```
    return Promise.all(imageRequests)
```

File: packages/11ty/_includes/web-components/image-sequence/index.js (hunk 2 above)
Comment (cbutcosk) [reply packages/11ty/_includes/web-components/image-sequence/index.js]: These are incorporated, see https://github.com/thegetty/quire/blob/ec1a79fc19adc1be430c3ea79ae12914079f1928/packages/11ty/_includes/web-components/image-sequence/index.js#L73

File: packages/11ty/_includes/web-components/image-sequence/index.js (hunk 3 above)
Comment (cbutcosk) [reply packages/11ty/_includes/web-components/image-sequence/index.js]: Incorporated (removed hideOverlays function and call in favor of relying on didInteract sentinel)

File: packages/11ty/_includes/web-components/image-sequence/index.js (hunk 4 above)
Comment (cbutcosk) [reply packages/11ty/_includes/web-components/image-sequence/index.js]: These changes are now incorporated (along with the switch of computing bufferWindow and bufferReady with private functions, and attendant variable renames, see https://github.com/thegetty/quire/blob/ec1a79fc19adc1be430c3ea79ae12914079f1928/packages/11ty/_includes/web-components/image-sequence/index.js#L404 )

File: packages/11ty/_includes/web-components/image-sequence/index.js (hunk 5 above)
Comment (cbutcosk) [reply packages/11ty/_includes/web-components/image-sequence/index.js]: See above, refined these to reflect your review

File: packages/11ty/_includes/web-components/image-sequence/index.js (hunk 6 above)
Comment (cbutcosk) [reply packages/11ty/_includes/web-components/image-sequence/index.js]: Also resolved along with the no-prefetch guard clause needing to return an empty Promise.all([])

**Review by mphstudios — COMMENTED** [review]
It looks like `#computeBufferReady` and `#computeBufferWindow` could both be written as implicit `#bufferReady` and `#bufferWindow` properties?

**Review by mphstudios — APPROVED** [review]

**Pull request merged**